Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/TanHelper.kt

```diff
@@ -0,0 +1,33 @@
+package de.rki.coronawarnapp.util
+
+import de.rki.coronawarnapp.ui.submission.TanConstants.MAX_LENGTH
+import java.security.MessageDigest
+import java.util.Locale
+
+object TanHelper {
+    private const val VALID_CHARACTERS = "23456789ABCDEFGHJKMNPQRSTUVWXYZ"
+
+    fun isChecksumValid(tan: String): Boolean {
+        if (tan.trim().length != MAX_LENGTH)
+            return false
+        val subTan = tan.substring(0, MAX_LENGTH - 1).toUpperCase(Locale.getDefault())
+        val tanDigest = MessageDigest.getInstance("SHA-256").digest(subTan.toByteArray())
```

Comment:
Converting strings to byte array without a charset is a bad idea. Please use `StandardCharsets.US_ASCII` here. Again, this is not an issue here, but this is bad code quality.

Reply:
Still worth fixing it ;) Thank you!